custom/enikshay/tasks.py

```diff
@@ -308,19 +309,54 @@ def update_json(self):
                 'expected_doses_taken': 0,
                 'aggregated_score_count_taken': 0,
                 'adherence_total_doses_taken': 0,
-                'adherence_latest_date_recorded': adherence_schedule_date_start - datetime.timedelta(days=1)
+                'adherence_latest_date_recorded': adherence_schedule_date_start - datetime.timedelta(days=1),
+                'one_week_score_count_taken': 0,
+                'two_week_score_count_taken': 0,
+                'month_score_count_taken': 0,
+                'one_week_adherence_score': 0,
+                'two_week_adherence_score': 0,
+                'month_adherence_score': 0,
             }
 
         adherence_cases = self.get_valid_adherence_cases()
         dose_taken_by_date = self.calculate_doses_taken_by_day(adherence_cases)
         update = self.get_aggregated_scores(
             latest_adherence_date, adherence_schedule_date_start, dose_taken_by_date)
+        update.update(self.get_adherence_scores(dose_taken_by_date))
 
         if self.check_if_needs_update(update):
             return update
         else:
             return None
 
+    def get_adherence_scores(self, doses_taken_by_date):
+        today = self.case_updater.date_today_in_india
+        start_date = self.get_adherence_schedule_start_date()
+        one_week_score_count_taken = self.count_doses_taken(
+            doses_taken_by_date,
+            start_date=today - datetime.timedelta(days=7),
+            end_date=today,
+        ) if today - datetime.timedelta(days=7) >= start_date else 0
```

Comment:
I think the `x = ... if ... else ...` construct is kinda hard to read when the clauses get long, breaking it up like this instead might be more readable:
```python
one_week_ago = today - datetime.timedelta(days=7)
if one_week_ago >= start_date:
    one_week_score_count_taken = self.count_doses_taken(
        doses_taken_by_date,
        start_date=one_week_ago,
        end_date=today,
    )
else:
    one_week_score_count_taken = 0
```